test/utils/test_skorch.py

```diff
@@ -0,0 +1,198 @@
+
+    # initialize dataset
+    dataset: Dataset = FakeDataset(
+        num_rows=30,
+        # with_nan=True,
+        stypes=stypes,
+        create_split=True,
+        task_type=task_type,
+        col_to_text_embedder_cfg=TextEmbedderConfig(
+            text_embedder=HashTextEmbedder(8)),
+    )
+    dataset.materialize()
+    train_dataset, val_dataset, test_dataset = dataset.split()
+    if not pass_dataset:
+        df_train = pd.concat([train_dataset.df, val_dataset.df])
+        X_train, y_train = df_train.drop(
+            columns=[dataset.target_col, dataset.split_col]), df_train[
+                dataset.target_col]
+        df_test = test_dataset.df
+        X_test, _ = df_test.drop(
+            columns=[dataset.target_col, dataset.split_col]), df_test[
+                dataset.target_col]
+
+        # never use dataset again
+        # we assume that only dataframes are available
+        del train_dataset, val_dataset, test_dataset
+
+    if cls == "mlp":
+        if module_as_function:
+
+            def get_module(col_stats: dict[str, dict[StatType, Any]],
+                           col_names_dict: dict[stype, list[str]]) -> MLP:
+                channels = 8
+                out_channels = 1
+                if task_type == TaskType.MULTICLASS_CLASSIFICATION:
+                    out_channels = dataset.num_classes
+                num_layers = 3
+                return MLP(
+                    channels=channels,
+                    out_channels=out_channels,
+                    num_layers=num_layers,
+                    col_stats=col_stats,
+                    col_names_dict=col_names_dict,
+                    normalization="layer_norm",
+                )
+
+            module = get_module
+            kwargs = {}
+        else:
+            module = MLP
+            kwargs = {
+                "channels":
+                8,
+                "out_channels":
+                dataset.num_classes
+                if task_type == TaskType.MULTICLASS_CLASSIFICATION else 1,
+                "num_layers":
+                3,
+                "normalization":
+                "layer_norm",
+            }
+            kwargs = {f"module__{k}": v for k, v in kwargs.items()}
+    else:
+        raise NotImplementedError
+    kwargs.update({
+        "module": module,
+        "criterion": loss,
+        "max_epochs": 2,
+        "verbose": 1,
+        "batch_size": 3,
+    })
+
+    if task_type == TaskType.REGRESSION:
+        net = NeuralNetPytorchFrame(**kwargs, )
+    if task_type == TaskType.MULTICLASS_CLASSIFICATION:
+        net = NeuralNetClassifierPytorchFrame(**kwargs, )
+    elif task_type == TaskType.BINARY_CLASSIFICATION:
+        net = NeuralNetBinaryClassifierPytorchFrame(**kwargs, )
+
+    if pass_dataset:
+        net.fit(dataset)
+        _ = net.predict(test_dataset)
+    else:
+        net.fit(X_train, y_train)
+        _ = net.predict(X_test)
```

Comment:
why don't we take tensor frame? It's also weird to sometimes take dataset and sometimes take data frame.

Reply:
- The main purpose of this PR is to allow DataFrame to be DIRECTLY fitted.
- Since it is unclear how to create a Dataset from a TensorFrame, and if there is a TensorFrame, there is also a Dataset, which means there is little need to use skorch.
- Instead of "sometimes take dataset and sometimes take data frame", both are tested.

Reply:
hmm if dataframe is directly fed, it is unclear why we need this feature within pytorch frame.
the whole point of pytorch frame is to materialize data frame into tensor frame, to be processed by pytorch.

Reply:
It may not match your purpose but my goal is to use advanced neural networks implemented in pytorch_frame in existing sklearn pipeline. 
This PR allows pytorch_frame to be used on top of existing scikit-learn code without having to heavily modify the existing code. Since many people use the sklearn pipeline, especially on Kaggle, it is easy to verify performance changes by changing or assembling the estimator in other people's code to my `NeuralNetPytorchFrame`. I am innovating this as a very win-win. I am convinced that this will be very valuable.